scipy/linalg/_decomp_qr.py

```diff
@@ -5,9 +5,15 @@
 from .lapack import get_lapack_funcs
 from ._misc import _datacopied
 
+from scipy._lib._array_api import array_namespace, is_numpy
+
 __all__ = ['qr', 'qr_multiply', 'rq']
 
 
+def arg_err_msg(param):
```

Comment:
will be renamed and imported from `_array_api.py` in a follow up to https://github.com/scipy/scipy/pull/19005.